src/v/cloud_storage/remote.h

```diff
@@ -455,8 +469,8 @@ class remote : public ss::peering_sharded_service<remote> {
         }
 
     private:
-        std::optional<std::reference_wrapper<retry_chain_node>>
-          _source_to_ignore;
+        fragmented_vector<std::reference_wrapper<retry_chain_node>>
+          _sources_to_ignore;
```

Comment:
I'm not sure it's a good idea to use fragmented vector with linear search here. We may have 1000 partitions per shard and every one of them will have some housekeeping job which means this vector will have 1000 elements to go through on every S3/ABS request. 

Maybe it makes sense to reverse the logic. Instead of storing sources that should be ignored you can store sources that should be included. This will be an ntp_archiver and all remote_segment instances. We have a limit on how many we can hydrate simultaneously which I believe is around 10.

Maybe this is not very convenient because `remote_segment` doesn't have access to the filter. So maybe it makes sense to just use another container. For instance, you can use the address of the source as a key.

Reply:
Hashing the address and using some sort of set should be good enough I think. Will update.

Reply:
Replaced with a hash set of pointers.